client/gutenberg/editor/calypsoify-iframe.jsx

```diff
@@ -29,16 +45,80 @@ const getIframeUrl = ( siteAdminUrl, postId, postType ) => {
 };
 
 class CalypsoifyIframe extends Component {
+	state = {
+		isMediaModalVisible: false,
+	};
+
+	constructor( props ) {
+		super( props );
+		this.iframeRef = React.createRef();
+	}
+
+	componentDidMount() {
+		window.addEventListener( 'message', this.onMessage, false );
+	}
+
+	componentWillUnmount() {
+		window.removeEventListener( 'message', this.onMessage, false );
+	}
+
+	onMessage = ( { data, origin, ports } ) => {
+		const message = parseJSON( data );
+		if ( ! message || origin.indexOf( this.props.siteSlug ) < 0 ) {
+			return;
+		}
+
+		const { action, type, payload } = message;
+
+		if ( 'gutenbergIframeMessage' !== type ) {
+			return;
+		}
+
+		if ( 'openMediaModal' === action && ports[ 0 ] ) {
+			this.messageChannelPort = ports[ 0 ];
```

Comment:
I didn't think much about the other side. where are we coding up the messages that we pass from inside the `iframe`?